src/arithmetic/montgomery.rs

```diff
@@ -152,14 +152,21 @@ pub(super) fn limbs_mul_mont(
             bn_mul_mont_ffi!(in_out, n, n0, (), unsafe {
                 (MIN_LIMBS, MOD_FALLBACK, ()) => bn_mul_mont_nohw
             })
-        } else if #[cfg(all(target_arch = "x86", target_feature = "sse2"))] {
-            // TODO: Do runtime detection of SSE2.
+        } else if #[cfg(target_arch = "x86")] {
+            use crate::{cpu::GetFeature as _, cpu::intel::Sse2};
             // The X86 implementation of `bn_mul_mont` has a minimum of 4.
             const _MIN_LIMBS_AT_LEAST_4: () = assert!(MIN_LIMBS >= 4);
-            let _: cpu::Features = cpu;
-            bn_mul_mont_ffi!(in_out, n, n0, (), unsafe {
-                (MIN_LIMBS, MOD_FALLBACK, ()) => bn_mul_mont
-            })
+            if let Some(cpu) = cpu.get_feature() {
+                bn_mul_mont_ffi!(in_out, n, n0, cpu, unsafe {
+                    (MIN_LIMBS, MOD_FALLBACK, Sse2) => bn_mul_mont
+                })
+            } else {
+                // This isn't really an FFI call; it's defined below.
+                unsafe {
+                    super::ffi::bn_mul_mont_ffi::<(), {MIN_LIMBS}, 1>(in_out, n, n0, (),
+                    bn_mul_mont_fallback)
+                }
```

Comment:
Codecov is right to complain that we don't test this branch in CI since we can't reach it without patching. This is intentional.